airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceMetadataQuerier.kt

```diff
@@ -0,0 +1,174 @@
+            base.conn.createStatement().use { stmt: Statement ->
+                stmt.executeQuery(sql).use { rs: ResultSet ->
+                    while (rs.next()) {
+                        results.add(
+                            AllPrimaryKeysRow(
+                                rs.getString("table_schema"),
+                                rs.getString("table_name"),
+                                rs.getString("constraint_name"),
+                                rs.getInt("ordinal_position").takeUnless { rs.wasNull() },
+                                rs.getString("column_name").takeUnless { rs.wasNull() },
+                            ),
+                        )
+                    }
+                }
+            }
+            log.info { "Discovered all primary keys in ${schemas.size} Mysql schema(s)." }
+            return@lazy results
+                .groupBy { findTableName(it.tableName, "public") }
+                .mapNotNull { (table, rowsByTable) ->
+                    if (table == null) return@mapNotNull null
+                    val pkRows: List<AllPrimaryKeysRow> =
+                        rowsByTable
+                            .groupBy { it.constraintName }
+                            .filterValues { rowsByPK: List<AllPrimaryKeysRow> ->
+                                rowsByPK.all { it.position != null && it.columnName != null }
+                            }
+                            .values
+                            .firstOrNull()
+                            ?: return@mapNotNull null
+                    val pkColumnNames: List<List<String>> =
+                        pkRows
+                            .sortedBy { it.position }
+                            .mapNotNull { it.columnName }
+                            .map { listOf(it) }
+                    table to pkColumnNames
+                }
+                .toMap()
+        } catch (e: Exception) {
+            throw RuntimeException("Mysql primary key discovery query failed: ${e.message}", e)
+        }
+    }
+
+    override fun primaryKey(
+        streamName: String,
+        streamNamespace: String?,
+    ): List<List<String>> {
+        val table: TableName = findTableName(streamName, streamNamespace) ?: return listOf()
+        return memoizedPrimaryKeys[table] ?: listOf()
+    }
+
+    private data class AllPrimaryKeysRow(
+        val tableSchema: String,
+        val tableName: String,
+        val constraintName: String,
+        val position: Int?,
+        val columnName: String?,
+    )
+
+    companion object {
+
+        const val PK_QUERY_FMTSTR =
+            """
+   SELECT 
+            table_schema,
+            table_name, 
+            column_name, 
+            ordinal_position, 
+            constraint_name 
+        FROM 
+            information_schema.key_column_usage 
+        WHERE 
+            table_schema IN (%s)
+            AND constraint_name = 'PRIMARY';
+            """
```

Comment:
Isn't this what the base JdbcMetadataQuerier is effectively querying? I'm guessing not, but I'm not seeing the difference. Can you explain what it is, please?

Reply:
I think the base could work too. I saw firetruck was implementing that so I simply overwrite it. Will test.